codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -518,7 +518,7 @@ public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promi
         public void onGoAwayRead(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData)
                 throws Http2Exception {
             // Don't allow any more connections to be created.
-            connection.goAwayReceived(lastStreamId);
+            connection.goAwayReceived(lastStreamId, errorCode, debugData);
```

Comment:
@nmittler I think it makes sense to have the `connection.goAwayReceived(...)` call first as this updates `goAwayReceived` and the `lastKnownStream`. It seems reasonable to do this first and only then notify external listeners.